src/components/portfolio/DetailsMainRisk.tsx

```diff
@@ -1,12 +1,251 @@
+
+// stylesheet for the risk analysis section
+const useStyles = makeStyles(({ palette }: Theme) =>
+  createStyles({
+    riskContainer: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      margin: '1rem 0',
+    },
+    ratioWrapper: {
+      width: '100%',
+      display: 'flex',
+      justifyContent: 'center',
+      margin: '4rem 0',
+    },
+    ratioText: {
+      width: 'min-content',
+      margin: '0 4rem',
+      padding: '1rem',
+      color: palette.primary.contrastText,
+      fontSize: '1.5rem',
+      fontWeight: 600,
+      whiteSpace: 'nowrap',
+      borderStyle: 'solid',
+      borderRadius: '0.75rem',
+      borderColor: palette.primary.contrastText,
+    },
+    riskWrapper: {
+      display: 'flex',
+      flexDirection: 'column',
+      width: '100%',
+    },
+    riskTitle: {
+      fontSize: '1.4rem',
+      margin: '0 0 1rem 0',
+      color: palette.primary.contrastText,
+    },
+    riskPieWrapper: {
+      margin: '1rem 0',
+      width: '100%',
+      alignSelf: 'center',
+    },
+    statContainer: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    iconWrapper: {
+      display: 'flex',
+      height: '4rem',
+      width: '4rem',
+    },
+    countWrapper: {
+      fontSize: '1.5rem',
+      marginRight: '1rem',
+    },
+    subtitleWrapper: {
+      width: '100%',
+      fontSize: '1.15rem',
+      fontWeight: 600,
+      color: palette.primary.contrastText,
+    },
+    icon: {
+      width: '100%',
+      height: '100%',
+    },
+    warnings: {
+      margin: '0.1rem 0',
+      fontSize: '1rem',
+      color: palette.primary.contrastText,
+    },
+  })
+);
+
+// type declarations
+type RiskCompProps = {
+  title: string;
+  bundle: RiskBundle;
+  labels: string[];
+  portions: number[];
+};
 
 type DetailsMainRiskProps = {
   risk: RiskAnalysis;
-  positions: Position[];
+  sharpeRatio: number;
+  treynorRatio: number;
+};
+
+// A component consisting of the title, chart and warnings of a given risk type
+const RiskComp: React.FC<RiskCompProps> = ({
+  title,
+  bundle,
+  labels,
+  portions,
+}) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  return (
+    <div className={classes.riskWrapper}>
+      {/* title */}
+      <div>
+        <p className={classes.riskTitle}>{title}</p>
+      </div>
+      {/* body with chart */}
+      <div className={classes.riskPieWrapper}>
+        <DetailsDonut
+          portions={portions}
+          labels={labels}
+          size={200}
+          graphOffsetX={-40}
+          showLegendOnScale={false}
+        />
+      </div>
+      {/* footer with warnings */}
+      <div>
+        {/* icon + number + title */}
+        <div className={classes.statContainer}>
+          <div className={classes.iconWrapper}>{bundle.riskIcon}</div>
+          <div className={classes.countWrapper}>
+            <div style={{ color: bundle.riskColor }}>{bundle.count}</div>
```

Comment:
I would use classes here instead of 'inline styles' here but I guess that's a matter of taste.